### ISSUE TITLE
 RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject

### ISSUE THREAD

**Issue by TrystanScottLambert** [astropy/astropy#11252]
### Description

When trying to import astropy.stats I get the above AttributeError. Making the code unusable. Downgrading numpy has not worked at all. 

### Expected behavior

astropy.stats to be imported. 

### Actual behavior

Crashed with the above error. Note that if I just import astropy.stats then I get a different error 

### Steps to Reproduce

```python
# Put your Python code snippet here.
from astropy.stats import sigma_clipped_stats

Traceback (most recent call last):
  File "apply_correction.py", line 52, in <module>
    from astropy.stats import sigma_clipped_stats
  File "/home/trystan/.local/lib/python3.6/site-packages/astropy/stats/__init__.py", line 14, in <module>
    from . import funcs
  File "/home/trystan/.local/lib/python3.6/site-packages/astropy/stats/funcs.py", line 17, in <module>
    from . import _stats
  File "__init__.pxd", line 918, in init astropy.stats._stats
RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 72, in apport_excepthook
    from apport.fileutils import likely_packaged, get_recent_crashes
  File "/usr/lib/python3/dist-packages/apport/__init__.py", line 5, in <module>
    from apport.report import Report
  File "/usr/lib/python3/dist-packages/apport/report.py", line 13, in <module>
    import fnmatch, glob, traceback, errno, sys, atexit, locale, imp, stat
  File "/usr/lib/python3.6/imp.py", line 33, in <module>
    DeprecationWarning, stacklevel=2)
DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses

Original exception was:
Traceback (most recent call last):
  File "apply_correction.py", line 52, in <module>
    from astropy.stats import sigma_clipped_stats
  File "/home/trystan/.local/lib/python3.6/site-packages/astropy/stats/__init__.py", line 14, in <module>
    from . import funcs
  File "/home/trystan/.local/lib/python3.6/site-packages/astropy/stats/funcs.py", line 17, in <module>
    from . import _stats
  File "__init__.pxd", line 918, in init astropy.stats._stats
RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
```

### System Details

[GCC 8.4.0]
Numpy 1.19.5
astropy 4.1
Scipy 1.5.4
Matplotlib 3.3.3

**Comment by pllim** [comment]
I am guessing this `apply_correction.py` is part of some package that has set to turn all warnings into errors? You could try to just ignore it like this:

https://github.com/astropy/astropy/blob/b3a328d7a061115fae108485a21299050f65b47c/setup.cfg#L121

But if you want to actually get rid of it, you might need to ensure that your Python environment isn't dirty. I see `/home/trystan/.local/lib/python3.6/site-packages/` and `/usr/lib/python3/dist-packages/` in your traceback, so not sure what's going on there.

Also, please try to reproduce it outside of your script in a Python session, and provide the System Details that we requested in the template. Thanks!

**Comment by TrystanScottLambert** [comment]
Funny enough what I've just noticed is that if I do everything in ipython via terminal first then it works and I can run my script. but just running the script right away causes the above errors and if I've attempted to run it once (via ipython) then I continue to get the same error.

**Comment by taldcroft** [comment]
BTW, if you are wondering about that warning there is useful info at:
-  https://stackoverflow.com/questions/40845304/runtimewarning-numpy-dtype-size-changed-may-indicate-binary-incompatibility
- https://github.com/numpy/numpy/issues/11788

tl;dr it should be benign. I've seen this and occasionally had to ignore it in pytest.ini to get warning-free test runs.

**Comment by TrystanScottLambert** [comment]
Thanks so much @taldcroft

**Comment by pllim** [comment]
@TrystanScottLambert ,  if your question has been answered, please close. I have marked as "Close?" for the bot to pick it up otherwise. Thanks!

**Comment by embray** [comment]
I get this sometimes; it's a Python environment error when you're using some extension modules that were compiled against a different version of Numpy than the one you're using.
